### DISCUSSION TITLE
Use citar to maintain two kinds of notes

### DISCUSSION THREAD

**Discussion by MArpogaus** [emacs-citar/citar#663] [Q&A]
Hello,

I would like to use citar to manage two types of notes:

### 1. General Notes
I would like to maintain multiple note files containing notes about the content of a book or paper i read.
It looks like the default notes function has been developed with this goal in mind and i just need to add a new template string.

```elisp
  (setq ctar-notes-template-multiple-files '(concat "#+TITLE: ${title}\n"
                                                   "#+AUTHOR: ${author editor}\n"
                                                   "#+DATE: ${date}\n"
                                                   "#+SOURCE: ${doi url}\n"
                                                   "#+CUSTOM_ID: ${=key= id}\n"
                                                   "#+cite_export: biblatex ieee\n"
                                                   (concat "#+bibliography: " citar-bibliography "\n\n")
                                                   "* Notes :ignore:\n"
                                                   ":PROPERTIES:\n"
                                                   ":NOTER_DOCUMENT: ${file}\n"
                                                   ":END:\n\n"
                                                   "* Summary :childless:showchildren:export:\n"
                                                   "This is a summary of [citet:@${=key=}].\n"
                                                   "** Bibliography :ignore:\n"
                                                   "#+print_bibliography "
                                                   ))
  (setq citar-bibliography (car (file-expand-wildcards (concat org-directory "bib/*.bib")))
        citar-notes-paths (list (concat org-directory "brain/bib_notes/"))
        citar-templates  `((main . "${author editor:30}     ${date year issued:4}     ${title:48}")
                          (suffix . "          ${=key= id:15}    ${=type=:12}    ${tags keywords:*}")
                          (preview . "${author editor} (${year issued date}) ${title}, ${journal journaltitle publisher container-title collection-title}.\n")
                          (note . ,(eval ctar-notes-template-multiple-files)))...)
```
In my first tests this seams to work pretty well.
The only problem i've discovered so far is that `${file}` yields all the files listed in the file attribute of the corresponding BibTeX entry.
Is there a easy way to make this selectable or just use the first occurring PDF file?

### 2. TODO Notes
Additionally, i would like to have a file containing todo entries for literature that I want to read.
Unfortunately i don't yet understand completely how to achieve that.
I thought of creating an other template like the following:

```elisp
  (setq citar-notes-template-one-file (concat "** TODO ${=key=}: ${title}, (${year}) :${type}:\n"
                                              ":PROPERTIES:\n"
                                              ":Custom_ID: ${=key=}\n"
                                              ":FILE: [[file:${file}]]\n"
                                              ":END:\n"
                                              "See [[cite:&${=key=}]]\n"))
```
an the define some custom notes function that uses this template an appends the content to a file.
I guess this function should the be added to `citar-open-note-functions` in order to chose between them in the embark menu.

Unfortunately i am very new to emacs and elist and it feels like this is far ahead from my current skill level.
I am very happy for any advice.

Thanks in advance
Marcel

**Comment by bdarcus** [comment]
Interesting.

> The only problem i've discovered so far is that ${file} yields all the files listed in the file attribute of the corresponding BibTeX entry.
Is there a easy way to make this selectable or just use the first occurring PDF file?

Not that I can think of. You'd probably need to write your own function for that.

BTW, you can play in IELM to see what's happening internally.

```elisp
ELISP> (citar-get-value "file" "testkey")
```

> Additionally, i would like to have a file containing todo entries for literature that I want to read.
Unfortunately i don't yet understand completely how to achieve that.

Instead of doing that, could you not create placeholder notes and have some status metadata for "TODO"/"DONE" "TOREAD/READ"? Org is designed for that kind of thing.

PS - we're working on this BTW https://github.com/emacs-citar/citar-org-roam

cc @roshanshariff

**Reply by MArpogaus** [reply]
Hello @roshanshariff,

thanks for your quick reply.

> Not that I can think of. You'd probably need to write your own function for that.

ok, do you know if the template itself may include elisp code like [org capture templates do](https://orgmode.org/manual/Template-expansion.html)?

Unfortunately i don't quite get what you mean by:

> Instead of doing that, could you not create placeholder notes and have some status metadata for "TODO"/"DONE" "TOREAD/READ"? Org is designed for that kind of thing.

**Reply by MArpogaus** [reply]
Looks like this function could be used to get all pdf files

    (citar-file--parse-file-field (citar--get-entry "KEY") "file" nil (list "pdf"))

**Reply by bdarcus** [reply]
I'm not Roshan; I was just tagging him ;-)

What I mean is:

https://orgmode.org/guide/TODO-Items.html

I guess in general, I don't think of TODO items as notes, and there are tons of features in org to manage projects, todo lists etc.

When I need to read something, for example, I put is as a TODO in a project-specific list of tasks. Like:

```org
** TODO Read [cite:doe22]
** DONE Write introduction
```

... and then use `embark-act` to open a new note at-point when you're ready.

It's just a thought to consider if you're new to Emacs and Org. I don't have a super specific suggestion.

**Reply by bdarcus** [reply]
> Looks like this function could be used to get all pdf files
> 
> ```
> (citar-file--parse-file-field (citar--get-entry "KEY") "file" nil (list "pdf"))
> ```

You can also try the more higher-level, more general:

```elisp
(citar-get-files "testkey")
```
